src/linker/Linker.Steps/ClearInitLocals.cs

```diff
@@ -30,28 +21,25 @@ private static IEnumerable<TypeDefinition> EnumerateTypesAndNestedTypes (IEnumer
 
 		protected override void ProcessAssembly (AssemblyDefinition assembly)
 		{
-			if (!Context.IsOptimizationEnabled (CodeOptimizations.ClearInitLocals, assembly) &&
-				_assemblies?.Contains (assembly.Name.Name) != true) {
+			if (!Context.IsOptimizationEnabled (CodeOptimizations.ClearInitLocals, assembly))
 				return;
-			}
 
-			bool changed = false;
+			if (Annotations.GetAction (assembly) != AssemblyAction.Link) {
+				Context.LogMessage (MessageContainer.CreateErrorMessage ($"Clearinitlocals is not supported for copy assembly '{assembly.Name.Name}'.", 1014));
```

Comment:
This can happen for any action (e.g. skip) so I'd not report it as error and just return